tqdm/_tqdm.py

```diff
@@ -133,10 +146,23 @@ def tqdm(iterable, desc=None, total=None, leave=False, file=sys.stderr,
     else:
         dynamic_miniters = False
 
+    if ascii is None:
+        try:
+            file.encoding
+        except AttributeError:
+            ascii = True
+        else:
+            if file.encoding:
+                ascii = not ('U8' == file.encoding or
+                             ('utf' in file.encoding) or
+                             ('UTF' in file.encoding))
+            else:
+                ascii = True
```

Comment:
What about extracting this to a helper function? Something like `file_suports_unicode`:

```
def _is_utf(encoding):
    return ('U8' == encoding) or ('utf' in encoding) or ('UTF' in encoding))

def file_supports_unicode(fp):
    if not getattr(fp, 'encoding', None):
        return False    
    return _is_utf(file.encoding)
```